modules/View.js

```diff
@@ -671,47 +589,209 @@
-      }
+    __activateTrackedViews: function(options) {
+      _.each(this.getTrackedViews(options), function(view) {
+        view.activate();
+      });
+    },
+
+    /**
+     * Disposes all child views recursively
+     * @method disposeChildViews
+     * @private
+     */
+    __disposeChildViews: function() {
       _.each(this.__childViews, function(view) {
-        view.invokeDetached();
+        view.dispose();
       });
-      _.each(this.__sharedViews, function(view) {
-        view.invokeDetached();
+    },
+
+    /**
+     * Will inject a new view into an injection site by using the new view's transitionIn method. If the parent view
+     * previously had another view at this injections site, this previous view will be removed with that view's transitionOut.
+     * If this method is used within a render, the current views' injection sites will be cached so they can be transitioned out even
+     * though they are detached in the process of re-rendering. If no previous view is given and none can be found, the new view is transitioned in regardless.
+     * If the previous view is the same as the new view, it is injected normally without transitioning in.
+     * The previous view must has used an injection site with the standard "inject=<name of injection site>" attribute to be found.
+     * @method transitionNewViewIntoSite
+     * @private
+     * @param injectionSiteName {String} The name of the injection site in the template. This is the value corresponding to the attribute "inject".
+     * @param newView {View} The instantiated view object to be transitioned into the injection site
+     * @param [options] {Object} optional options object. This options object will be passed on to the transitionIn and transitionOut methods as well.
+     * @param   [options.previousView] {View} the view that should be transitioned out. If none is provided, it will look to see if a view already
+     *                                 is at this injection site and uses that by default.
+     * @param   [options.addBefore=false] {Boolean} if true, the new view's element will be added before the previous view's element. Defaults to after.
+     * @param   [options.shared=false] {Boolean} if set to true, the view will be treated as a shared view and not disposed during parent view disposing.
+     * @return {Promise} resolved when all transitions are complete. No payload is provided upon resolution.
+     * When the transitionIn and transitionOut methods are invoked on the new and previous views, the options parameter will be passed on to them. Other fields
+     * will be added to the options parameter to allow better handling of the transitions. These include:
+     * {
+     *   newView: the new view
+     *   previousView: the previous view (can be undefined)
+     *   parentView: the parent view transitioning in or out the tracked view
+     * }
+     */
+    __transitionNewViewIntoSite: function(injectionSiteName, newView, options) {
```

Comment:
this method feels long, does it make sense to have private methods for readability and make it clear what the major steps of the method are?